code/modules/mob/living/silicon/ai/ai.dm

```diff
@@ -316,7 +316,9 @@
 	var/turf/target = get_turf(A)
 	if (.)
 		return
-	if ((ai.z != target.z) && !is_station_level(ai.z))
+
+	//Not all things the AI clicks on may have a source turf, so we check if target exists for this check (examples include obj/screen items)
+	if ((target && ai.z != target.z) && !is_station_level(ai.z))

```

Comment:
Looking at the rest of this proc, shouldn't you just return false right away if target is null?

Reply:
no, because previously it was just feeding directly into the cameranet turf vis procs and I haven't audited that for how it handles something that doesn't actually have a turf.

it's possible it just returns, but I don't want to modify the behaviour unnecessarily here.

Reply:
It's not feeding the (A)tom into those procs, it's feeding in get_turf(A) which is null

Looking in the proc itself, that will runtime every time since it expects to be given something not null. Unless I'm missing something obvious?

Reply:
I'll trust you and adjust this when I remember to get to it